### ISSUE TITLE
ExceptionInInitializerError while AccessToken.getToken

### ISSUE THREAD

**Issue by hzy00** [facebook/facebook-android-sdk#612]
## Environment

- Java Version: 1.8
- Android SDK Version: 28
- gradle:3.3.1
- facebook sdk: [5,6) actually 5.8.0
- firebase-core: 16.0.7 
- firebase-crash: 16.2.1 (Somebody said facebook not compatible with firebase)

## Goals

launch application successful

## Expected Results

launch application successful

## Actual Results

application crash

## Code Samples & Details

I've set `AutoLogAppEventsEnabled` false, and init sdk manually.
```
<meta-data
      android:name="com.facebook.sdk.AutoLogAppEventsEnabled"
      android:value="false"/>

FirebaseApp.initializeApp(getApplicationContext());
FacebookSdk.setApplicationId(BuildConfig.FACEBOOK_APP_ID);
Looper.myQueue().addIdleHandler(() -> {
  FacebookSdk.sdkInitialize(getApplicationContext());
};
```

Facebook sdk crash stacks:
```
Fatal Exception: java.lang.ExceptionInInitializerError
com.facebook.internal.FetchedAppSettingsManager$1.run (FetchedAppSettingsManager.java:202)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1112)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
java.lang.Thread.run (Thread.java:818)
Caused by java.lang.NullPointerException
Attempt to read from field 'java.lang.String com.facebook.a.e' on a null object reference
com.facebook.AccessToken.getToken (AccessToken.java:235)
com.facebook.appevents.AccessTokenAppIdPair.<init> (AccessTokenAppIdPair.java:35)
com.facebook.appevents.AppEventsLoggerImpl.<init> (AppEventsLoggerImpl.java:565)
com.facebook.appevents.AppEventsLoggerImpl.<init> (AppEventsLoggerImpl.java:544)
com.facebook.appevents.InternalAppEventsLogger.<init> (InternalAppEventsLogger.java:48)
com.facebook.appevents.internal.AutomaticAnalyticsLogger.<clinit> (AutomaticAnalyticsLogger.java:59)
com.facebook.internal.FetchedAppSettingsManager$1.run (FetchedAppSettingsManager.java:202)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1112)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
java.lang.Thread.run (Thread.java:818)
```
Any idea please?

**Comment by hzy00** [comment]
AnyBody here？

**Comment by olegpetsjonkin** [comment]
SDK version 5.8 - same erro happens. Investigation revealed race condition in SDK:
FacebookSdk.java:

```
... 
261: public static synchronized void sdkInitialize(
...
312:  FetchedAppSettingsManager.loadAppSettingsAsync(); (1)
...
347:  FutureTask<Void> futureTask =
...
351: AccessTokenManager.getInstance().loadCurrentAccessToken(); (2)
```
(2) asynchronously loads accessToken
(1) FetchedAppSettingsManager.java:
```
...
157:  FacebookSdk.getExecutor().execute(new Runnable() {
...
202: AutomaticAnalyticsLogger.logActivateAppEvent(); (3)
...
```
(3) fast forward -> AutomaticAnalyticsLogger -> InternalAppEventsLogger -> AppEventsLoggerImpl.java :
```
...
558: accessToken = AccessToken.getCurrentAccessToken(); (4)
...
562: if (AccessToken.isCurrentAccessTokenActive() &&  (5)
...
565: accessTokenAppId = new AccessTokenAppIdPair(accessToken); (6)
...
```
If (4) is executed first, then (2), then (5) -> NPE in (6)
To reporduce: 
1. put breakpoints in (2), (4), (5)
2. debugger stops on (2) and (4) - release (4)
3. debugger stopped on (2) and stops on (5) - release (2)
4. release (5)
NPE
To fix: 
AppEventsLoggerImpl.java : 
line 562 : change to : 
```
if (accessToken != null && !accessToken.isExpired() && 
```

**Comment by hzy00** [comment]
@KylinChang @jingping2015 anybody could help to solve the problem?

**Comment by hzy00** [comment]
Too terrible! Every week, This crash is high in Top3 for my app! however, nobody care about it here!

**Comment by KylinChang** [comment]
Hi @hzy00 , sorry for the late reply, the fix is shipped in v5.11.0 as you can see in https://github.com/facebook/facebook-android-sdk/blob/master/CHANGELOG.md. I suggest reporting a bug in https://developers.facebook.com/bugs/ so that we can react more quickly. Close the issue now and feel free to reopen if the issue still exists.

**Comment by hzy00** [comment]
OKay, thanks for your reply. I'll track it in my next app version.
